recipes/samarium/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2022 Jai Bellare
+# See <https://opensource.org/licenses/MIT/> or LICENSE.md
+# Project homepage: <https://github.com/strangeQuark1041/samarium>
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class SamariumConan(ConanFile):
+    name = "samarium"
+    version = "1.0.0"
+    description = "2-D physics simulation library"
+    homepage = "https://strangequark1041.github.io/samarium/"
+    url = "https://github.com/conan-io/conan-center-index/"
+    license = "MIT"
+    topics = ("c++20", "physics", "2d", "simulation")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    generators = "cmake_find_package"
```

Comment:
Only cmake_find_package generator is not enough, need a wrapper to set all cmake definitions